lib/util/http/index.ts

```diff
@@ -80,6 +81,25 @@ function applyDefaultHeaders(options: Options): void {
   };
 }
 
+async function gotTask<T>(
```

Comment:
Can this refactoring be done in a separate PR now?

Reply:
We don't have `.then(...)` calls in `master` branch. Or do you mean something else?

Reply:
I was wondering how much of the refactoring we can do separately before adding the queue. Or do you think it's not really possible?

Reply:
I thought this function looked like it could be refactored independently of the queuing

Reply:
Let's see what can I extract...